feat(no-duplicate-attr-inheritance): ignore multi root (#2598)
vuejs/eslint-plugin-vue: docs/rules/no-duplicate-attr-inheritance.md

## :book: Rule Details

> This rule aims to prevent duplicate attribute inheritance.
> This rule to warn to apply `inheritAttrs: false` when it detects `v-bind="$attrs"` being used.

Reviewer: ```suggestion
This rule aims to prevent duplicate attribute inheritance.  
This rule suggests applying `inheritAttrs: false` when it detects `v-bind="$attrs"` being used.
```

  /* ✓ GOOD */
  inheritAttrs: false
}

> </script>

Reviewer: Good catch!

  /* ✗ BAD */
  // inheritAttrs: true (default)
}
</script>
```

</eslint-code-block>

> ### `"checkMultiRootNodes": true`

Reviewer: This heading should be below the options.

<eslint-code-block :rules="{'vue/no-duplicate-attr-inheritance': ['error', { checkMultiRootNodes: true }]}">

```vue
<template>
  <div v-bind="$attrs" />
  <div />
</template>
<script>
export default {
  /* ✗ BAD */
  // inheritAttrs: true (default)
}
</script>
```

</eslint-code-block>

## :wrench: Options

```json
{
  "vue/no-duplicate-attr-inheritance": ["error", {
    "checkMultiRootNodes": false,
  }]
}
```

> - `"checkMultiRootNodes"` ... If `true`, check and warn when there are multiple root nodes. (Default: `false`)

Reviewer: Can you please add an explanation why the default is `false` (and that is fine), and why someone might still want to enable the option?

Author: Add more information and link to the Vue documentation for further explanation.

  /* ✗ BAD */
  // inheritAttrs: true (default)
}
</script>
```

</eslint-code-block>

## :wrench: Options

```json
{
  "vue/no-duplicate-attr-inheritance": ["error", {
    "checkMultiRootNodes": false,
  }]
}
```

> - `"checkMultiRootNodes"`: If set to `true`, check and warn to apply `inheritAttrs: false` whenever it detects `v-bind="$attrs"` being used. Default is `false`, will ignore the components with multiple root nodes, because Vue encourages adding `v-bind="$attrs"` in this case to prevent runtime warnings. See [attribute inheritance on multiple root nodes](https://vuejs.org/guide/components/attrs.html#attribute-inheritance-on-multiple-root-nodes) for more.

Reviewer: ```suggestion
- `"checkMultiRootNodes"`: If set to `true`, also suggest applying `inheritAttrs: false` to components with multiple root nodes (where `inheritAttrs: false` is the implicit default, see [attribute inheritance on multiple root nodes](https://vuejs.org/guide/components/attrs.html#attribute-inheritance-on-multiple-root-nodes)), whenever it detects `v-bind="$attrs"` being used. Default is `false`, which will ignore components with multiple root nodes.
```
